widgets/matplotlib/simulation/composition.py

```diff
@@ -223,6 +245,31 @@ def __update_selected_layer(self):
 
         self.canvas.draw_idle()
 
+    def __update_reference_density(self, layers: List):
+
+        reference_density = 0.0
+        layer_reference_density = 0.0
+        reference_density_limit = 10.0
+        masses = []
+        g_mass = 0.0
+
+        for layer in layers:
+            if reference_density_limit - layer.start_depth > 0:
+                for element in layer.elements:
+                    mass = element.get_mass()
+                    masses.append(mass * element.amount)
+                    for mass in masses:
+                        g_mass += (gf.convert_amu_to_kg(mass) * 1000)
+                    layer_density = layer.density
+                    layer_reference_density += (layer_density / g_mass)
+                reference_density = layer_reference_density * (
+                        layer.thickness / reference_density_limit)
+
+        self.parent.referenceDensityLabel.setText(
+            f"Reference density: "
+            f"{reference_density:1.2e} "
+            f"at./cm\xb3")
+
```

Comment:
This seems like a mistake: it's not called anywhere, and its code is roughly contained in `ReferenceDensity.calculate_reference_density()` and `on_draw()`.

Reply:
Yes, probably a mistake